tests/Unit/ConsoleTest/ConsoleOutputTest.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+use Peck\Console\Commands\CheckCommand;
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Exception\InvalidOptionException;
+use Symfony\Component\Console\Tester\CommandTester;
+
+it('requires a config value when --config is passed', function (): void {
+    // Find the application
+    $application = new Application;
+    $application->add(new CheckCommand);
+    $command = $application->find('check');
+
+    // Execute the command, but don't pass a config file
+    (new CommandTester($command))->execute([
+        '--config' => null,
+    ]);
+})->throws(InvalidOptionException::class);
+
+it('throws an exception when an invalid --config is passed', function (): void {
+    // Find the application
+    $application = new Application;
+    $application->add(new CheckCommand);
+    $command = $application->find('check');
+
+    // Execute the command, but don't pass a config file
+    (new CommandTester($command))->execute([
+        '--config' => 'i-dont-exist',
+    ]);
+})->throws(InvalidOptionException::class);
+
+it('works when a valid --config is passed', function (): void {
+    // Arrange
+
+    // I've chosen to create a small config file here, as opposed to creating files all over the place, & losing track.
+    $tempConfig = 'peck2.json';
```

Comment:
Maybe you could take a look at the `tmpfile()` Function ([Docs](https://www.php.net/manual/en/function.tmpfile.php)).

It makes sure that you get a temporary file, which is cleaned up after usage and the file name is automatically unique.

Reply:
Using `tmpfile()` would be the ideal situation here, but `Config.php`  forces us to use a local "config" file
```php
$basePath = dirname(array_keys(ClassLoader::getRegisteredLoaders())[0]);
$filePath = $basePath.'/'.(self::$resolveConfigFilePathUsing instanceof Closure
    ? (self::$resolveConfigFilePathUsing)()
    : 'peck.json');
```
So when I tried it, It never worked. I worked it back to this part of the code which came up with the `$filePath` `home/john/Projects/peck//tmp/randStr`
So I opted for the touch/unlink option.
I'm obviously open to a more suitable method.